metrics.py

```diff
@@ -6,22 +6,23 @@
 TARGET_WIDTH = 74
 N = K.cast(TARGET_HEIGHT * TARGET_WIDTH, dtype='float32')
 
-# set y true to small non zero number if it's zero to avoid division error
-# set y true to 1 if infinity
-def clean_y_true(y_true):
-    y_true = tf.where(y_true == 0, 0.001, y_true)
-    y_true = tf.where(tf.math.is_inf(y_true), 1.0, y_true)
-    return y_true
+# set y true to 1 if it's zero to avoid NaN division error
+# set y pred to 1 if y true was 0 (y true becomes 1 after this block) so
+# that the metric does not take it into account (difference = 0)
+def clean_y_true(y_true, y_pred):
+    y_pred = tf.where(y_true == 0, 1.0, y_pred)
```

Comment:
I have a slight worry that the float equality comparison doesn't work, instead we might need to do the epsilon thing.

but, let try this first. LGTM!